pycaret/internal/plots/time_series.py

```diff
@@ -812,18 +812,12 @@ def plot_time_series_decomposition(
     decomp_result = None
     data_ = data.to_timestamp() if isinstance(data.index, pd.PeriodIndex) else data
 
-    sp_to_use = data_kwargs.get("sp_to_use", None)
+    sp_to_use = data_kwargs.get("sp_to_use")
     if plot == "decomp_classical":
-        if sp_to_use is None:
-            decomp_result = seasonal_decompose(data_, model=classical_decomp_type)
-        else:
             decomp_result = seasonal_decompose(
                 data_, period=sp_to_use, model=classical_decomp_type
             )
     elif plot == "decomp_stl":
-        if sp_to_use is None:
-            decomp_result = STL(data_).fit()
-        else:
             decomp_result = STL(data_, period=sp_to_use).fit()
```

Comment:
Could you please confirm that this works when `sp_to_use = 1`. If so, then this looks good. Thanks!

Reply:
As in the classical decomposition, since `sp_to_use` cannot be None, L825 was unreachable code really. But in this case, the STL decomposition does not work with `sp_to_use = 1`: statsmodels code does not allow it (and it makes sense, because it would be pointless otherwise):
```
statsmodels/tsa/_stl.pyx in statsmodels.tsa._stl.STL.__init__()
ValueError: period must be a positive integer >= 2
```
Note also that this is already happening, it is no new behavior. I'd leave this behavior as is: it makes sense for this to fail if `sp_to_use = 1` and the error is pretty informative.

Reply:
Fair enough. This is a bug though and we will need to fix this. The appropriate action would be to through a warning message in this case from pycaret rather than have the program crash. I will open a new issue for this.

Thanks for catching this!